server/channels/store/sqlstore/property_group_store.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	sq "github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+)
+
+var propertyGroupColumns = []string{"ID", "Name"}
+
+type SqlPropertyGroupStore struct {
+	*SqlStore
+}
+
+func newPropertyGroupStore(sqlStore *SqlStore) store.PropertyGroupStore {
+	return &SqlPropertyGroupStore{sqlStore}
+}
+
+func (s *SqlPropertyGroupStore) Register(name string) (*model.PropertyGroup, error) {
+	if name == "" {
+		return nil, store.NewErrInvalidInput("PropertyGroup", "name", name)
+	}
+
+	existingGroup, err := s.Get(name)
```

Comment:
Once we have the `ON CONFLICT` clause, then we don't really need this. Because we are already doing:

```
if rowsAffected == 0 {
		return s.Get(name)
	}
```
later. So this can be removed.

Reply:
Right, good catch, removed here 4a54403d0a